lib/flop/schema.ex

```diff
@@ -333,6 +332,56 @@ defprotocol Flop.Schema do
           as: :pet_count
         )
         |> Flop.validate_and_run(params, for: Owner)
+
+  ## Custom fields
+
+  If you need more control over the queries produced by the filters you can define
+  custom fields that reference a function, which implements the filter logic.
+  Custom fields filters are referenced by `{mod :: module(), function :: atom(), opts :: keyword()}`.
+  The function will receive the Ecto query, the flop filter and the option keyword list.
+
+  If you need to pass in options at runtime (e.g. the timezone of the request, the user id of the current user etc)
+  you can do so by passing in the `extra_opts` option to the flop functions.
+  Currently custom fields only support filtering and can not be used for sorting.
+
+  Schema:
+
+      @derive {
+        Flop.Schema,
+        filterable: [:inserted_at_date],
+        custom_fields: [inserted_at_date: [filter: {CustomFilters, :date_filter, [source: :inserted_at]}]]
+      }
+
+  Filter module:
+
+      defmodule CustomFilters do
+        def date_filter(query, %Flop.Filter{value: value, op: op}, opts) do
+          source = opts[:source]
+          timezone = opts[:timezone]
```

Comment:
I know it's only an example, but...

```suggestion
          source = Keyword.fetch!(opts, :source)
          timezone = Keyword.fetch!(opts, :timezone)
```